pkg/model/manifest.go

```diff
@@ -260,9 +260,11 @@ func (m1 Manifest) fieldGroupsEqual(m2 Manifest) (primitivesEq, dockerEq, k8sEq,
 	lt2 := m2.LocalTarget()
 	localEq = DeepEqual(lt1, lt2)
 
-	depsEq = DeepEqual(m1.ResourceDependencies, m2.ResourceDependencies)
+	depsEq = sliceutils.StringSliceElementsMatch(m1.LocalPaths(), m2.LocalPaths())
```

Comment:
I don't understand this check? 

LocalPaths() are derived from the other data models, so e.g., if dockerEq is false, than depsEq is false. Right?

Reply:
relatedly, I feel like we should be moving Manifest towards using the DeepEqual infrastructure, rather than adding more manually curated checks

Reply:
in `DeepEquals` (which as you point out is a v questionable name) i ignore a handful of fields (`CustomBuild.Deps` and `LocalTarget.Deps` that are ONLY relevant to `LocalPaths()` and nothing else; if one of those fields changes, we want it to NOT invalidate the build (but we still say the manifests are unequal), hence splitting out the `depsEq` (so if all that's different btwn m1 and m2 is `CustomBuild.Deps`, then `dockerEq = true` ("everything but the deps") and `depsEq = false`).

For something like `DockerBuild.BuildPath`, it affects `LocalPaths` but also other stuff (it's used in the image build), so a change in that field DOES mean that `dockerEq = false`. does that make sense/sound reasonable?

Reply:
hmmmm....i think i see your point, but...it's going to make this pretty complicated to reason about.

TBH, I'm pretty sure this is all moot, i sent you this PR:
https://github.com/tilt-dev/tilt/pull/3745
which removes the Equal() method. Then that will make a bunch of this complexity disappear, since the return values will be unused